SQL Anywhere: Wrong DDL syntax `CREATE SEQUENCE ... AS ...`

### Search first

- [X] I searched and no similar issues were found

### Description

Using latest SQL Anywhere 17.0.11-7254 as a *standalone* engine (**without** using testcontainers) I am trying to execute integration tests, and this is the result:
```
[ERROR] liquibase.dbtest.asany.SybaseASAIntegrationTest.testRerunDiffChangeLog  Time elapsed: 24.851 s  <<< ERROR!
liquibase.exception.CommandExecutionException:
liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset changelogs/common/common.tests.changelog.xml::seq-test-datatype::flautert:
     Reason: liquibase.exception.DatabaseException: [SAP][JDBC Driver][SQL Anywhere]Syntaxfehler bei 'AS' in Zeile 1  [Failed SQL: (-131) CREATE SEQUENCE lbuser.seqtestdatatypeonall AS int]
```
Apparently Liquibase expects that SQL Anywhere supports the syntax `CREATE SEQUENCE ... AS ...`, but according to the [latest product manual](https://help.sap.com/docs/SAP_SQL_Anywhere/93079d4ba8e44920ae63ffb4def91f5b/3be47d496c5f1014bf06bb7791ce7a38.html?locale=en-US&q=CREATE%20SEQUENCE) SQL Anywhere is (and I assume, it ever was) unable to deal with the `AS ...` clause:
>CREATE [ OR REPLACE ] SEQUENCE [ owner .] sequence-name
[ INCREMENT BY signed-integer ]
[ START WITH signed-integer ]
[ MINVALUE signed-integer | NO MINVALUE ]
[ MAXVALUE signed-integer | NO MAXVALUE ]
[ CACHE integer | NO CACHE ]
[ CYCLE | NO CYCLE ]

### Steps To Reproduce

```bash
dbinit -dba lbuser,LiquibasePass1 liquibase
```
```liquibase.sdk.local.yaml
liquibase:
  sdk:
    testSystem:
      test: asany

      asany:
        url: jdbc:sqlanywhere:DBF=liquibase.db
```
```bash
mvn verify -Dtest=SybaseASAIntegrationTest -Dsurefire.failIfNoSpecifiedTests=false
```


### Expected/Desired Behavior

The IT passes successfully.

### Liquibase Version

master

### Database Vendor & Version

SAP SQL Anywhere 17.0.11

### Liquibase Integration

N/A

### Liquibase Extensions

N/A

### OS and/or Infrastructure Type/Provider

Windows 10 Pro (German)

### Additional Context

Standalone engine / Without Testcontainers

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)

--- a/liquibase-standard/src/main/java/liquibase/sqlgenerator/core/CreateSequenceGenerator.java
+++ b/liquibase-standard/src/main/java/liquibase/sqlgenerator/core/CreateSequenceGenerator.java
@@ -99,8 +99,10 @@ public class CreateSequenceGenerator extends AbstractSqlGenerator<CreateSequence
         if (statement.getCacheSize() != null) {
             if (database instanceof OracleDatabase || database instanceof Db2zDatabase || database instanceof PostgresDatabase || database instanceof MariaDBDatabase || database instanceof SybaseASADatabase) {
                 if (BigInteger.ZERO.equals(statement.getCacheSize())) {
-                    if (database instanceof OracleDatabase || database instanceof SybaseASADatabase) {
+                    if (database instanceof OracleDatabase) {
                         queryStringBuilder.append(" NOCACHE ");
+                    } else if (database instanceof SybaseASADatabase) {
+                        queryStringBuilder.append(" NO CACHE ");
                     } else if (database instanceof MariaDBDatabase) {
                         queryStringBuilder.append(" CACHE 0");
                     }

--- a/liquibase-standard/src/main/java/liquibase/sqlgenerator/core/CreateSequenceGenerator.java
+++ b/liquibase-standard/src/main/java/liquibase/sqlgenerator/core/CreateSequenceGenerator.java
@@ -76,7 +76,7 @@ public class CreateSequenceGenerator extends AbstractSqlGenerator<CreateSequence
         if (database instanceof HsqlDatabase || database instanceof Db2zDatabase) {
             queryStringBuilder.append(" AS BIGINT ");
         } else if (statement.getDataType() != null) {
-            if (!(isH2WithoutAsDatatypeSupport(database) || database instanceof CockroachDatabase)) {
+            if (!(isH2WithoutAsDatatypeSupport(database) || database instanceof CockroachDatabase) || database instanceof SybaseASADatabase) {
                 queryStringBuilder.append(" AS ").append(statement.getDataType());
             }
         }
@@ -97,9 +97,9 @@ public class CreateSequenceGenerator extends AbstractSqlGenerator<CreateSequence
         }
 
         if (statement.getCacheSize() != null) {
-            if (database instanceof OracleDatabase || database instanceof Db2zDatabase || database instanceof PostgresDatabase || database instanceof MariaDBDatabase) {
+            if (database instanceof OracleDatabase || database instanceof Db2zDatabase || database instanceof PostgresDatabase || database instanceof MariaDBDatabase || database instanceof SybaseASADatabase) {
                 if (BigInteger.ZERO.equals(statement.getCacheSize())) {
-                    if (database instanceof OracleDatabase) {
+                    if (database instanceof OracleDatabase || database instanceof SybaseASADatabase) {
                         queryStringBuilder.append(" NOCACHE ");
                     } else if (database instanceof MariaDBDatabase) {
                         queryStringBuilder.append(" CACHE 0");

--- a/liquibase-standard/src/main/java/liquibase/sqlgenerator/core/CreateSequenceGenerator.java
+++ b/liquibase-standard/src/main/java/liquibase/sqlgenerator/core/CreateSequenceGenerator.java
@@ -76,7 +76,7 @@ public class CreateSequenceGenerator extends AbstractSqlGenerator<CreateSequence
         if (database instanceof HsqlDatabase || database instanceof Db2zDatabase) {
             queryStringBuilder.append(" AS BIGINT ");
         } else if (statement.getDataType() != null) {
-            if (!(isH2WithoutAsDatatypeSupport(database) || database instanceof CockroachDatabase) || database instanceof SybaseASADatabase) {
+            if (!(isH2WithoutAsDatatypeSupport(database) || database instanceof CockroachDatabase || database instanceof SybaseASADatabase)) {
                 queryStringBuilder.append(" AS ").append(statement.getDataType());
             }
         }

--- a/liquibase-standard/src/test/java/liquibase/sqlgenerator/core/CreateSequenceGeneratorTest.java
+++ b/liquibase-standard/src/test/java/liquibase/sqlgenerator/core/CreateSequenceGeneratorTest.java
@@ -4,6 +4,7 @@ import liquibase.database.Database;
 import liquibase.database.DatabaseConnection;
 import liquibase.database.core.H2Database;
 import liquibase.database.core.PostgresDatabase;
+import liquibase.database.core.SybaseASADatabase;
 import liquibase.exception.DatabaseException;
 import liquibase.exception.ValidationErrors;
 import liquibase.sql.Sql;
@@ -16,6 +17,8 @@ import java.math.BigInteger;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.Mockito.*;
+import static java.math.BigInteger.ONE;
+import static java.math.BigInteger.ZERO;
 
 public class CreateSequenceGeneratorTest extends AbstractSqlGeneratorTest<CreateSequenceStatement> {
 
@@ -32,6 +35,25 @@ public class CreateSequenceGeneratorTest extends AbstractSqlGeneratorTest<Create
     @Override
     protected CreateSequenceStatement createSampleSqlStatement() {
         return new CreateSequenceStatement(CATALOG_NAME, SCHEMA_NAME, SEQUENCE_NAME);
+    }
+
+    @Test
+    public void sqlanywhereDatabaseSupportNoCache() {
+        // given
+        DatabaseConnection dbConnection = mock(DatabaseConnection.class);
+        SybaseASADatabase database = spy(new SybaseASADatabase());
+        database.setConnection(dbConnection);
+        doReturn(SEQUENCE_NAME).when(database).escapeSequenceName(CATALOG_NAME, SCHEMA_NAME, SEQUENCE_NAME);
+        CreateSequenceStatement createSequenceStatement = createSampleSqlStatement();
+        createSequenceStatement.setStartValue(ONE);
+
+        // when
+        createSequenceStatement.setCacheSize(ZERO);
+
+        // then
+        Sql[] sql = new CreateSequenceGenerator().generateSql(createSequenceStatement, database, new MockSqlGeneratorChain());
+        assertThat(sql).isNotEmpty().hasSize(1);
+        assertThat(sql[0].toSql()).contains("NO CACHE");
     }
 
     @Test

